### PR TITLE
[DROOLS-6531] fix OR of 2 accumulates in executable model

### PR DESCRIPTION
**Pull request by mariofusco**
**JIRA**: https://issues.redhat.com/browse/DROOLS-6531

### CODE REVIEW

File: drools-model/drools-model-compiler/src/main/java/org/drools/modelcompiler/KiePackagesBuilder.java [hunk 1]
```diff
@@ -690,12 +691,18 @@ private ConditionalBranch buildConditionalConsequence(RuleContext ctx, Condition
...
         for (int i = 0; i < subconditions.size(); i++) {
             RuleConditionElement element = conditionToElement( ctx, ge, subconditions.get(i) );
             if (element != null) {
                 ge.addChild( element );
             }
         }
+        if (ge.getType() == OR) {
+            ctx.endOrCondition();
+        }
         if (ge.getType() == AND && ge.getChildren().size() == 1) {
```
Comment (pibizza) [inline drools-model/drools-model-compiler/src/main/java/org/drools/modelcompiler/KiePackagesBuilder.java:706]: Is it possible to have "or" conditions at multiple levels? I am wondering if the end can be called when we have a complex condition.

**Review by pibizza — COMMENTED** [review]
As a general comment we miss unit tests. I understand that many of these classes have no unit test, and they require lots of initial effort to be put in a unit test fixture. On the other side, once the fixture is in place adding another test is typically quite simple.

I suggest to identify the classes that we know we are going to modify a lot, and add fixtures and initial unit tests for them.

File: drools-model/drools-model-compiler/src/main/java/org/drools/modelcompiler/KiePackagesBuilder.java (hunk 1 above)
Comment (mariofusco) [reply drools-model/drools-model-compiler/src/main/java/org/drools/modelcompiler/KiePackagesBuilder.java:706]: @pibizza it is possible to have nested ORs and then I added in the `RuleContext` class a stack of List of Variables to take count of this situation.

**Review by lucamolteni — APPROVED** [review]

File: drools-model/drools-model-compiler/src/main/java/org/drools/modelcompiler/KiePackagesBuilder.java (hunk 1 above)
Comment (pibizza) [reply drools-model/drools-model-compiler/src/main/java/org/drools/modelcompiler/KiePackagesBuilder.java:706]: Ok, of course I didn't notice that. Thanks. Looks good to me.

**Review by pibizza — APPROVED** [review]

**Review by hellowdan — APPROVED** [review]

**Pull request merged**